### PR TITLE
Change oltpreceiver to use the new unmarshaler, avoid grpc-gateway dependency

### PR DESCRIPTION
**Pull request by bogdandrutu**
Updates https://github.com/open-telemetry/opentelemetry-collector/issues/3104

### CODE REVIEW

File: receiver/otlpreceiver/otlp.go [hunk 1]
```diff
@@ -16,11 +16,12 @@ package otlpreceiver
 
 import (
 	"context"
+	"io/ioutil"
 	"net"
 	"net/http"
 	"sync"
 
-	gatewayruntime "github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/gorilla/mux"
```
Comment (jrcamp) [inline receiver/otlpreceiver/otlp.go:24]: grpc-gateway changes should maybe be in separate PR?

Comment (bogdandrutu) [reply receiver/otlpreceiver/otlp.go:24]: https://github.com/open-telemetry/opentelemetry-collector/pull/3422

**Review by jrcamp — APPROVED** [review]
